src/cloud/index.js

```diff
@@ -0,0 +1,193 @@
+
+  await deleteDeployment(environment, service, subservice);
+}
+
+export async function info(options) {
+  const { environment, service, subservice } = options;
+  if (!devMode) assertBedrockRoot();
+
+  const config = await getConfig(environment);
+  await checkConfig(environment, config);
+  const deployment = getDeployment(service, subservice);
+  const deploymentInfo = await checkDeployment(service, subservice);
+  const { annotations } = deploymentInfo.spec.template.metadata;
+  console.info(`Deployment "${deployment}" annotations:`);
+  console.info(annotations);
+}
+
+export async function provision(options) {
+  const { environment, terraform } = options;
+  if (!devMode) assertBedrockRoot();
+
+  const config = await getConfig(environment);
+  await checkGCloudProject(config.gcloud);
+
+  await provisionTerraform(environment, terraform, config.gcloud);
+}
+
+export async function shell(options) {
+  const { environment, service, subservice } = options;
+  if (!devMode) assertBedrockRoot();
+
+  const config = await getConfig(environment);
+  await checkConfig(environment, config);
+
+  const podsJSON = await exec(`kubectl get pods -o jsonpath='{.items}' --ignore-not-found`);
+  if (!podsJSON) {
+    console.info(kleur.yellow(`No running pods`));
+    process.exit(0);
+  }
+  const pods = JSON.parse(podsJSON.slice(1, -1));
+
+  let deployment = 'api-cli-deployment';
+  if (service) {
+    deployment = getDeployment(service, subservice);
+  }
+
+  const filteredPods = pods.filter((pod) => pod.metadata.name.startsWith(deployment));
+
+  if (!filteredPods.length) {
+    console.info(kleur.yellow(`No running pods for deployment "${deployment}"`));
+    process.exit(0);
+  }
+
+  const podName = filteredPods[0].metadata.name;
+  console.info(kleur.green(`=> Starting bash for pod: "${podName}"`));
+
+  const { spawn } = require('child_process');
+
+  const child = spawn('kubectl', ['exec', '-it', podName, '--', 'bash'], {
+    stdio: 'inherit',
+  });
+
+  child.on('exit', function (code) {
+    console.info(kleur.green(`Finished bash for pod: "${podName}" (exit code: ${code})`));
+  });
+}
+
+export async function logs(options) {
+  const { environment, service, subservice } = options;
+  if (!devMode) assertBedrockRoot();
+
+  const config = await getConfig(environment);
+  await checkConfig(environment, config);
+  const { project, computeZone, kubernetes, label } = config.gcloud;
+
+  let labelName = service;
+  if (subservice) labelName += `-${subservice}`;
+  const podLabel = label || 'app';
+
+  const query = `resource.type="k8s_container"\nresource.labels.project_id="${project}"\nresource.labels.location="${computeZone}"\nresource.labels.cluster_name="${kubernetes.clusterName}"\nresource.labels.namespace_name="default"\nlabels.k8s-pod/${podLabel}="${labelName}"`;
+
+  const params = new URLSearchParams({ query });
+
+  const url = `https://console.cloud.google.com/logs/query;${params.toString()}?project=${project}`;
```

Comment:
this is a very odd url... is this how it's actually structured?

Reply:
Don't get me started, ask Google :) This is the URL I see when I'm browsing the Google Logs Explorer.